src/devices/Mcp3008/Mcp3008.cs

```diff
@@ -77,36 +77,31 @@ public int Read(int channel, InputConfiguration inputConfiguration = InputConfig
             }
         }
 
-        private static byte GetConfigurationBits(int channel,InputConfiguration inputConfiguration = InputConfiguration.SingleEnded)
+        private static byte GetConfigurationBits(int channel, InputConfiguration inputConfiguration)
         {
-            int configurationBits;
+            int configurationBits = (0x18 | channel) << 3;
 
-            if (inputConfiguration == InputConfiguration.SingleEnded)
+            if (inputConfiguration == InputConfiguration.Differential)
             {
-                configurationBits = 0b00011000 | channel;
-            }
-            else
-            {
-                configurationBits = 0b00010000 | channel;
+                configurationBits &= 0xBF;  // Clear mode bit.
```

Comment:
NIT: For things like these I usually prefer something like `configurationBits &= ~(0b1u << 6)` so that it is more obvious which bit are you clearing

Reply:
If we are going to use binary format, then it seems like bit shift would get a little weird (especially if you have multiple bits to set/rotate.  In this example, you can clearly see which bit your trying to check/clear.

`configurationBits &= 0b1011_1111;  // 0xBF`